build.gradle

```diff
@@ -0,0 +1,248 @@
+	testClassesDirs = testing.suites.test.sources.output.classesDirs
+	classpath = testing.suites.test.sources.runtimeClasspath
+
+	filter {
+		excludeTestsMatching "*LongTest"
+	}
+	finalizedBy "jacocoTestQuickReport" // must be a string, see TODOs at head of file
+}
+
+task jacocoTestQuickReport(type: JacocoReport, dependsOn: testQuick) {
+	sourceSets sourceSets.main
+	executionData(testQuick)
+}
+
+// Custom java project layout
+sourceSets {
+	main {
+		java {
+			srcDirs = ["src"]
+			exclude '**/test/**'
+		}
+	}
+	test {
+		java {
+			srcDirs = ["src"]
+			include '**/test/**'
+		}
+		resources {
+			srcDirs = ["src"]
+			include "**/*.properties"
+			include '**/*.spec'
+			include '**/*instance.*'
+			include '**/*.instance*'
+			include '**/*.html'
+			include '**/*.css'
+			include '**/*.gif'
+			include '**/*.js'
+			include '**/*.png'
+			include '**/*.txt'
+			include '**/*.weirdsuffix'
+		}
+	}
+}
+
+javadoc {
+	options {
+		links "https://docs.oracle.com/en/java/javase/11/docs/api/"
+		links "https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-core/2.9.9/"
+		links "https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/2.9.9/"
+	}
+}
+
+war {
+	webXml = file('war/web.xml')
+}
+
+configurations {
+	// can't directly access testImplementation, so extend and access
+	testimpl.extendsFrom testImplementation
+}
+
+def fromURL = { url, name ->
+	File file = new File("$buildDir/download/${name}.jar")
+	file.parentFile.mkdirs()
+	if (!file.exists()) {
+		new URL(url).withInputStream { downloadStream ->
+			file.withOutputStream { fileOut ->
+				fileOut << downloadStream
+			}
+		}
+	}
+	files(file.absolutePath)
+}
+
+dependencies {
+
+	// ### General application dependencies ###
+	
+	implementation fromURL(
+		'https://github.com/kbase/jars/raw/master/lib/jars/kbase/auth/kbase-auth-0.4.4.jar',
+		'kbase-auth-0.4.4'
+	)
+	implementation fromURL(
+		'https://github.com/kbase/jars/raw/master/lib/jars/kbase/kidl/kbase-kidl-parser-1409261812-7863aef.jar',
+		'kbase-kidl-parser-1409261812-7863aef'
+	)
+	implementation 'com.fasterxml.jackson.core:jackson-annotations:2.9.9'
+	implementation 'com.fasterxml.jackson.core:jackson-core:2.9.9'
+	implementation 'com.fasterxml.jackson.core:jackson-databind:2.9.9'
+	implementation 'org.ini4j:ini4j:0.5.2'
+	implementation 'commons-io:commons-io:2.4'
+	implementation 'org.apache.commons:commons-lang3:3.1'
+	implementation 'commons-codec:commons-codec:1.8'
+	implementation 'info.picocli:picocli:4.6.1'
+	implementation 'org.mongodb:mongodb-driver-core:4.11.1'
+	implementation 'org.mongodb:mongodb-driver-sync:4.11.1'
+	implementation 'org.mongodb:bson-record-codec:4.11.1'
+	implementation 'org.mongodb:bson:4.11.1'
+	implementation 'org.slf4j:slf4j-api:1.7.30'
+	implementation 'ch.qos.logback:logback-classic:1.1.2'
+	implementation 'com.google.guava:guava:14.0.1'
+	implementation 'com.github.ben-manes.caffeine:caffeine:2.9.3'
+	implementation 'org.apache.kafka:kafka-clients:2.1.0'
```

Comment:
can you alphabetise this list?

Reply:
done. I put all the jars repo deps first and then alphabetized each subsection. Re the subsections see comment below